packages/noir-ethereum-api/src/noir/oracles/accountOracles.ts

```diff
@@ -40,3 +55,25 @@ export const getAccountOracle = async (client: PublicClient, args: string[][]):
     encodeField(account.nonce),
   ];
 }
+
+export function convertEthDataToNoirProof(block: EthBlock, proof: EthProof): AccountWithProof {
```

Comment:
You probably don't need to encode both at the same time, there are two oracles:
- header - returns header
- account - returns account with proof
Noir uses both and validates one against the other.